### DISCUSSION TITLE
How to run AVA tests with a pre-transpile step

### DISCUSSION THREAD

**Discussion by sebastianrath** [avajs/ava#3087] [Q&A] [answered]
Hey there,

I am using `ava` on a project that contains more and more tests. It's time to consider speeding up the warm-up time for test files which now take a considerable amount of time.

My project is written in TypeScript, and so are the tests. I used a configuration with `ts-node` to execute the tests. This still looks like the way to go when executing individual test files within VScode.

But tests in a pipeline could benefit from a pre-transpiled test suite, and I am unable to find much information about it.

The documentation about [AVA/TypeScript](https://github.com/avajs/ava/blob/main/docs/recipes/typescript.md) only refers to [ts-node]. Is there a recommended way to run the tests through AVA when test files are already transpiled in the output directory `./build`?

Any advice is highly appreciated.

**Comment by novemberborn** [comment, marked as answer]
> Is there a recommended way to run the tests through AVA when test files are already transpiled in the output directory ./build?

Yes: https://github.com/avajs/ava/blob/main/docs/recipes/typescript.md#with-precompile-step

See https://github.com/avajs/typescript.

```
{
	"ava": {
		"typescript": {
			"rewritePaths": {
				"src/": "build/"
			},
			"compile": false
		}
	}
}
```

**Reply by sebastianrath** [reply]
Ah, I misunderstood the documentation. Thanks for the quick reply.
